src/executables/util_cs.c

```diff
@@ -3994,6 +3994,798 @@ tde (UTIL_FUNCTION_ARG * arg)
+
+  for (i = 0; i < num_cond_col; i++)
+    {
+      ptr = or_unpack_int (ptr, &cond_index);
+    }
+
+  attr = class_->ordered_attributes;
+  for (i = 0; i < num_cond_col; i++)
+    {
+      error = check_sql_memory (&sql, strlen (sql) + strlen (attr->header.name) + 12, &max_sql_size);
+      if (error != NO_ERROR)
+	{
+	  goto error;
+	}
+
+      process_column_data (&ptr, &sql, &max_sql_size, attr->type->id);
+      if (i != num_cond_col - 1)
+	{
+	  strcat (sql, ", ");
+	}
+      else
+	{
+	  strcat (sql, " )");
+	}
+
+      attr = attr->order_link;
+    }
+
+  if (is_detail)
+    {
+      ptr = *loginfo;
+
+      original_sql = (char *) malloc (max_original_size);
+      if (original_sql == NULL)
+	{
+	  error = ER_OUT_OF_VIRTUAL_MEMORY;
+	  goto error;
+	}
+      sprintf (original_sql, "delete from %s where ", classname);
+
+      ptr = or_unpack_int (ptr, &num_change_col);
+      ptr = or_unpack_int (ptr, &num_cond_col);
+
+      for (i = 0; i < num_cond_col; i++)
+	{
+	  ptr = or_unpack_int (ptr, &cond_index);
+	}
+
+      attr = class_->ordered_attributes;
+      for (i = 0; i < num_cond_col; i++)
+	{
+	  error = check_sql_memory (&sql, strlen (original_sql) + strlen (attr->header.name) + 12, &max_sql_size);
+	  if (error != NO_ERROR)
+	    {
+	      goto error;
+	    }
+
+	  sprintf (original_sql + strlen (original_sql), "%s = ", attr->header.name);
+
+	  process_column_data (&ptr, &original_sql, &max_original_size, attr->type->id);
+
+	  if (i != num_cond_col - 1)
+	    {
+	      strcat (original_sql, " and ");
+	    }
+	  else
+	    {
+	      strcat (original_sql, " limit 1");
+	    }
+
+	  attr = attr->order_link;
+	}
+    }
+
+  *loginfo = ptr;
+
+  if (is_detail)
+    {
+      print_flashback_detail (trid, user, sql, original_sql, outfp);
+    }
+  else
+    {
+      fprintf (outfp, "%s\n", sql);
+    }
+
+  free_and_init (sql);
+  if (original_sql != NULL)
+    {
+      free_and_init (original_sql);
+    }
+
+  return NO_ERROR;
+
+error:
+  if (sql != NULL)
+    {
+      free_and_init (sql);
+    }
+
+  if (original_sql != NULL)
+    {
+      free_and_init (original_sql);
+    }
+
+  return error;
+}
+
+static int
+print_flashback_insert (char **loginfo, int trid, char *user, const char *classname, bool is_detail, FILE * outfp)
+{
+  MOP classop;
+  SM_CLASS *class_;
+  SM_ATTRIBUTE *attr;
+
+  int error = NO_ERROR;
+
+  char *sql = NULL;
+  int max_sql_size = 1024;
+
+  char *original_sql = NULL;
+  int max_original_size = 1024;
+
+  char *ptr;
+
+  int num_change_col;
+  int change_index;
+
+  int num_cond_col;
+
+  int i = 0;
+
+  sql = (char *) malloc (max_sql_size);
+  if (sql == NULL)
+    {
+      error = ER_OUT_OF_VIRTUAL_MEMORY;
+      goto error;
+    }
+
+  ptr = *loginfo;
+
+  classop = sm_find_class (classname);
+  error = au_fetch_class (classop, &class_, AU_FETCH_READ, AU_SELECT);
+  if (error != NO_ERROR)
+    {
+      goto error;
+    }
+
+  sprintf (sql, "delete from %s where ", classname);
```

Comment:
The table name should be enclosed by [ and ]. And maybe other identifiers. The following test is successfully executed, unfortunately.
- create table [test 2] (c1 int);
- insert into [test 2] valus (1);
- select * from [test 2];

But, an sql fails to execute if there is no [ and ].

Reply:
Should we support an additional utility option for that? What do guys(@joohok @wrlawodms @eido5) think about it?

Reply:
All object name including column's and table's has to be enclosed with []. It lets the names be distinguished with reserved keywords such as 'table' and 'create'. For example, we can do `create table [table] (a int);` with this. @hornetmj You mean an option that enables printing []? It looks necessary to me.

Reply:
I agree with you. 
I didn't consider about it. I will fix it

Reply:
OK. I agree with the opinion of @wrlawodms.